src/PixiEditor/Views/MainWindow.xaml.cs

```diff
@@ -247,4 +249,13 @@ private void MainWindow_OnKeyDown(object sender, KeyEventArgs e)
             e.Handled = true;
         }
     }
+
+    private void Viewport_OnContextMenuOpening(object sender, ContextMenuEventArgs e)
+    {
+        if (DataContext.ToolsSubViewModel.RightClickMode != RightClickMode.ContextMenu &&
+            DataContext.ToolsSubViewModel.ActiveTool is ShapeTool)
```

Comment:
I'd argue that the context menu should be fully disabled on other modes, instead of it being based on the tool (open for discussion tho)